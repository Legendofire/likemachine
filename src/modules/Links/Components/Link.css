.link__container{
  box-sizing: border-box;
  display: grid;
  grid-template: 1.5fr 4fr 2fr / 1fr;
  width: 100%;
  height: 100%;
  min-height: 300px;
  box-shadow: 0px 0px 2px rgba(0,0,0,0.3);
}

.link__title{
  display: flex;
  flex-direction: row;
  justify-content:space-between;
  align-items: center;
  padding-left: 4px;
  padding-right: 1px;
  width: 100%;
}

.link__title h3{
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 240px;
}

.likes__container{
  height: 50px;
  width: 90px;
  display: flex;
  flex-direction: row;
  align-items: center;
}

.heart{
  height: 40px;
  width: 40px;
  filter: grayscale(100%);
}

.heart--liked{
  filter: none;
}

.link__image{
  max-width: 100%;
  max-height: 100%;
  background-color: rgba(0,0,0,0.1);
}

.link__image img{
  width: 100%;
  height: 100%;
  object-fit: contain;
}

.link__description{
  box-sizing: border-box;
  max-width: 100%;
  max-height: 100%;
  text-overflow: ellipsis;
  overflow-y: auto;
  padding: 5px;
}
