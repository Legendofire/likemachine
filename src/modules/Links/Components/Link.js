import React, { Component } from "react";
import PropTypes from 'prop-types';

import "./Link.css";

export default class Link extends Component{
  render(){
    return(
      <div className="link__container">
        <div className="link__title">
          <h3>{this.props.title}</h3>
          <div className="likes__container">
            <a onClick={this.props.liked?this.props.unlikeAction:this.props.likeAction}>
              <img className={this.props.liked?"heart heart--liked":"heart"} src="heart.png" alt=""/>
            </a>
            <h3>{this.props.count}</h3>
          </div>
        </div>
        <a href={this.props.url} onClick={this.props.openUrlAction} className="link__image">
          <img src={this.props.src} alt=""/>
        </a>
        <div className="link__description">
          <h5>{this.props.description}</h5>
        </div>
      </div>
    )
  }
}

Link.propTypes = {
  title: PropTypes.string,
  likeAction: PropTypes.func,
  unLikeAction: PropTypes.func,
  liked: PropTypes.bool,
  count: PropTypes.number,
  url: PropTypes.string,
  openUrlAction: PropTypes.func,
  src: PropTypes.string,
  description: PropTypes.string
};
