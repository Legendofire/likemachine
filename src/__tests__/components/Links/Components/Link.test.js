import React from 'react';
import renderer from 'react-test-renderer';

import Link from '../../../../modules/Links/Components/Link';

test('Link Renders With Supplied Data', () => {
  const component = renderer.create(
    <Link key={1}
      liked={true}
      count={302}
      title={"This is a title"}
      description={"This is a description"}
      url={"http://www.google.com"}
      src={"http://www.facebook.com"}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
