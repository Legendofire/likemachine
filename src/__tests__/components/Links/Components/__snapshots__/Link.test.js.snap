// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Renders With Supplied Data 1`] = `
<div
  className="link__container"
>
  <div
    className="link__title"
  >
    <h3>
      This is a title
    </h3>
    <div
      className="likes__container"
    >
      <a
        onClick={undefined}
      >
        <img
          alt=""
          className="heart heart--liked"
          src="heart.png"
        />
      </a>
      <h3>
        302
      </h3>
    </div>
  </div>
  <a
    className="link__image"
    href="http://www.google.com"
    onClick={undefined}
  >
    <img
      alt=""
      src="http://www.facebook.com"
    />
  </a>
  <div
    className="link__description"
  >
    <h5>
      This is a description
    </h5>
  </div>
</div>
`;
